app/javascript/toolbar-actions/util.spec.js

```diff
@@ -0,0 +1,43 @@
+import { convertMultParamsToRailsMultParams } from './util';
+
+const TESTCASES = [
+  [ // Example Case
+    "selectedUser=all&approvalStateCheckboxes=pending_approval&approvalStateCheckboxes=approved&approvalStateCheckboxes=denied&types=all&selectedPeriod=7&reasonText=",
+    "selectedUser=all&approvalStateCheckboxes[]=pending_approval&approvalStateCheckboxes[]=approved&approvalStateCheckboxes[]=denied&types=all&selectedPeriod=7&reasonText="
+  ], [
+    "foo=bar",
+    "foo=bar"
+  ], [
+    "foobar",
+    "foobar=undefined"
+  ], [
+    "foo=bar&tree&color=red",
+    "foo=bar&tree=undefined&color=red"
+  ], [ // key named: brok&en
+    "var1=8&var2=5&brok&en=value",
+    "var1=8&var2=5&brok=undefined&en=value"
+  ], [ // value named: brok&en
+    "var1=8&var2=5&var1=brok&en",
+    "var1[]=8&var1[]=brok&var2=5&en=undefined"
+  ], [ // key named: brok=en
+    "var1=8&var2=5&brok=en=value",
+    "var1=8&var2=5&brok=en"
+  ], [ // value named: brok=en
+    "var1=8&var2=5&var1=brok=en",
+    "var1[]=8&var1[]=brok&var2=5"
```

Comment:
Same here I'd expect

```suggestion
    "var1=8&var2=5&var1=brok=en",
    "var1[]=8&var1[]=brok%3Den&var2=5"
```